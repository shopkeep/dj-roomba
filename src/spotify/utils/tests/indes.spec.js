const { convertToSongObject, getTracks } = require("../");
const songMock = require("./test-data/song");
const tracksMock = require("./test-data/tracks");

describe("spotify utils", function() {
  describe("convertToSongObject", function() {
    it("converts repsonse to song object", function() {
      expect(convertToSongObject(songMock)).toMatchSnapshot();
    });
  });

  describe("getTracks", function() {
    it("extracts playable tracks", function() {
      expect(getTracks(tracksMock)).toMatchSnapshot();
    });
  });
});
