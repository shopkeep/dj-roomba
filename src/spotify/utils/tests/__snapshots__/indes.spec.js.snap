// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spotify utils convertToSongObject converts repsonse to song object 1`] = `
Object {
  "album": Object {
    "id": "30Perjew8HyGkdSmqguYyg",
    "image": Object {
      "height": 640,
      "url": "https://i.scdn.co/image/5a6a1c6514398dc4004c6348a83d77694a3883d4",
      "width": 640,
    },
    "name": "Blood Sugar Sex Magik (Deluxe Version)",
  },
  "artist": "Red Hot Chili Peppers",
  "duration": 264306,
  "explicit": false,
  "id": "3d9DChrdc6BOeFsbrZ3Is0",
  "name": "Under The Bridge",
  "popularity": 82,
  "url": "https://open.spotify.com/track/3d9DChrdc6BOeFsbrZ3Is0",
}
`;

exports[`spotify utils getTracks extracts playable tracks 1`] = `
Array [
  Object {
    "album": Object {
      "album_type": "album",
      "artists": Array [
        Object {
          "external_urls": Object {
            "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
          },
          "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
          "id": "0L8ExT028jH3ddEcZwqJJ5",
          "name": "Red Hot Chili Peppers",
          "type": "artist",
          "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
        },
      ],
      "external_urls": Object {
        "spotify": "https://open.spotify.com/album/30Perjew8HyGkdSmqguYyg",
      },
      "href": "https://api.spotify.com/v1/albums/30Perjew8HyGkdSmqguYyg",
      "id": "30Perjew8HyGkdSmqguYyg",
      "images": Array [
        Object {
          "height": 640,
          "url": "https://i.scdn.co/image/5a6a1c6514398dc4004c6348a83d77694a3883d4",
          "width": 640,
        },
        Object {
          "height": 300,
          "url": "https://i.scdn.co/image/5ec4ff29311f065813234bdeee7c1364960f0c3e",
          "width": 300,
        },
        Object {
          "height": 64,
          "url": "https://i.scdn.co/image/0fe6b80fd539eb74825b58c3fa62db459fa76b68",
          "width": 64,
        },
      ],
      "name": "Blood Sugar Sex Magik (Deluxe Version)",
      "release_date": "1991-09-24",
      "release_date_precision": "day",
      "type": "album",
      "uri": "spotify:album:30Perjew8HyGkdSmqguYyg",
    },
    "artists": Array [
      Object {
        "external_urls": Object {
          "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
        },
        "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
        "id": "0L8ExT028jH3ddEcZwqJJ5",
        "name": "Red Hot Chili Peppers",
        "type": "artist",
        "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
      },
    ],
    "disc_number": 1,
    "duration_ms": 264306,
    "explicit": false,
    "external_ids": Object {
      "isrc": "USWB19901576",
    },
    "external_urls": Object {
      "spotify": "https://open.spotify.com/track/3d9DChrdc6BOeFsbrZ3Is0",
    },
    "href": "https://api.spotify.com/v1/tracks/3d9DChrdc6BOeFsbrZ3Is0",
    "id": "3d9DChrdc6BOeFsbrZ3Is0",
    "is_local": false,
    "is_playable": true,
    "name": "Under The Bridge",
    "popularity": 82,
    "preview_url": "https://p.scdn.co/mp3-preview/90e41778392f27b6f7dd82db4c90916b3727aa6a?cid=2e581de7a4de43ea90e2ede13f73f090",
    "track_number": 11,
    "type": "track",
    "uri": "spotify:track:3d9DChrdc6BOeFsbrZ3Is0",
  },
  Object {
    "album": Object {
      "album_type": "compilation",
      "artists": Array [
        Object {
          "external_urls": Object {
            "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
          },
          "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
          "id": "0L8ExT028jH3ddEcZwqJJ5",
          "name": "Red Hot Chili Peppers",
          "type": "artist",
          "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
        },
      ],
      "external_urls": Object {
        "spotify": "https://open.spotify.com/album/53tvjWbVNZKd3CvpENkzOC",
      },
      "href": "https://api.spotify.com/v1/albums/53tvjWbVNZKd3CvpENkzOC",
      "id": "53tvjWbVNZKd3CvpENkzOC",
      "images": Array [
        Object {
          "height": 640,
          "url": "https://i.scdn.co/image/55cc31d3efa495d05e034146331553db0a747cf5",
          "width": 640,
        },
        Object {
          "height": 300,
          "url": "https://i.scdn.co/image/a6cdb66f1035edb25b045ddc4fcce7908baa043d",
          "width": 300,
        },
        Object {
          "height": 64,
          "url": "https://i.scdn.co/image/35f5e13e4e9b1953f03534def9fa879dc202eb50",
          "width": 64,
        },
      ],
      "name": "Greatest Hits",
      "release_date": "2003-11-18",
      "release_date_precision": "day",
      "type": "album",
      "uri": "spotify:album:53tvjWbVNZKd3CvpENkzOC",
    },
    "artists": Array [
      Object {
        "external_urls": Object {
          "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
        },
        "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
        "id": "0L8ExT028jH3ddEcZwqJJ5",
        "name": "Red Hot Chili Peppers",
        "type": "artist",
        "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
      },
    ],
    "disc_number": 1,
    "duration_ms": 265506,
    "explicit": false,
    "external_ids": Object {
      "isrc": "USWB19901576",
    },
    "external_urls": Object {
      "spotify": "https://open.spotify.com/track/23NPGXlSaIqWzvxIRhM2oG",
    },
    "href": "https://api.spotify.com/v1/tracks/23NPGXlSaIqWzvxIRhM2oG",
    "id": "23NPGXlSaIqWzvxIRhM2oG",
    "is_local": false,
    "is_playable": true,
    "name": "Under The Bridge",
    "popularity": 65,
    "preview_url": "https://p.scdn.co/mp3-preview/5c052c976f9ad6610a1e8daf79e467e1e3943632?cid=2e581de7a4de43ea90e2ede13f73f090",
    "track_number": 1,
    "type": "track",
    "uri": "spotify:track:23NPGXlSaIqWzvxIRhM2oG",
  },
  Object {
    "album": Object {
      "album_type": "compilation",
      "artists": Array [
        Object {
          "external_urls": Object {
            "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
          },
          "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
          "id": "0L8ExT028jH3ddEcZwqJJ5",
          "name": "Red Hot Chili Peppers",
          "type": "artist",
          "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
        },
      ],
      "external_urls": Object {
        "spotify": "https://open.spotify.com/album/6kLm0YrEESYq7yszohzkA1",
      },
      "href": "https://api.spotify.com/v1/albums/6kLm0YrEESYq7yszohzkA1",
      "id": "6kLm0YrEESYq7yszohzkA1",
      "images": Array [
        Object {
          "height": 640,
          "url": "https://i.scdn.co/image/a0995e38da4b8b1b6a4b97c44d6cba12af079171",
          "width": 640,
        },
        Object {
          "height": 300,
          "url": "https://i.scdn.co/image/1fc78fb83e0eeb55488cb3553bd6eb3959b2ced0",
          "width": 300,
        },
        Object {
          "height": 64,
          "url": "https://i.scdn.co/image/4d6e47c29579fa529db5d4d4c8da794d9b0a40e9",
          "width": 64,
        },
      ],
      "name": "Greatest Hits",
      "release_date": "2003-11-18",
      "release_date_precision": "day",
      "type": "album",
      "uri": "spotify:album:6kLm0YrEESYq7yszohzkA1",
    },
    "artists": Array [
      Object {
        "external_urls": Object {
          "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
        },
        "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
        "id": "0L8ExT028jH3ddEcZwqJJ5",
        "name": "Red Hot Chili Peppers",
        "type": "artist",
        "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
      },
    ],
    "disc_number": 1,
    "duration_ms": 265506,
    "explicit": false,
    "external_ids": Object {
      "isrc": "USWB19901576",
    },
    "external_urls": Object {
      "spotify": "https://open.spotify.com/track/4zkzkEQGjfzmq0tVzpsMdM",
    },
    "href": "https://api.spotify.com/v1/tracks/4zkzkEQGjfzmq0tVzpsMdM",
    "id": "4zkzkEQGjfzmq0tVzpsMdM",
    "is_local": false,
    "is_playable": true,
    "name": "Under The Bridge",
    "popularity": 39,
    "preview_url": "https://p.scdn.co/mp3-preview/5c052c976f9ad6610a1e8daf79e467e1e3943632?cid=2e581de7a4de43ea90e2ede13f73f090",
    "track_number": 1,
    "type": "track",
    "uri": "spotify:track:4zkzkEQGjfzmq0tVzpsMdM",
  },
  Object {
    "album": Object {
      "album_type": "compilation",
      "artists": Array [
        Object {
          "external_urls": Object {
            "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
          },
          "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
          "id": "0L8ExT028jH3ddEcZwqJJ5",
          "name": "Red Hot Chili Peppers",
          "type": "artist",
          "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
        },
      ],
      "external_urls": Object {
        "spotify": "https://open.spotify.com/album/7aOFj1yM42sm2gzmlPidPP",
      },
      "href": "https://api.spotify.com/v1/albums/7aOFj1yM42sm2gzmlPidPP",
      "id": "7aOFj1yM42sm2gzmlPidPP",
      "images": Array [
        Object {
          "height": 640,
          "url": "https://i.scdn.co/image/e6947bf08107a5f990e5b6a9989ad90d99472c2e",
          "width": 640,
        },
        Object {
          "height": 300,
          "url": "https://i.scdn.co/image/5d923faa3a4df99d6ce5b603fb2b200d3e908b46",
          "width": 300,
        },
        Object {
          "height": 64,
          "url": "https://i.scdn.co/image/828e3ded367beb594c95a8420263be0b9a4e88c5",
          "width": 64,
        },
      ],
      "name": "The Studio Album Collection 1991 - 2011",
      "release_date": "2014-12-12",
      "release_date_precision": "day",
      "type": "album",
      "uri": "spotify:album:7aOFj1yM42sm2gzmlPidPP",
    },
    "artists": Array [
      Object {
        "external_urls": Object {
          "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
        },
        "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
        "id": "0L8ExT028jH3ddEcZwqJJ5",
        "name": "Red Hot Chili Peppers",
        "type": "artist",
        "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
      },
    ],
    "disc_number": 1,
    "duration_ms": 265506,
    "explicit": false,
    "external_ids": Object {
      "isrc": "USWB19901576",
    },
    "external_urls": Object {
      "spotify": "https://open.spotify.com/track/1MMlmA1APa0MevW3Bm2PpE",
    },
    "href": "https://api.spotify.com/v1/tracks/1MMlmA1APa0MevW3Bm2PpE",
    "id": "1MMlmA1APa0MevW3Bm2PpE",
    "is_local": false,
    "is_playable": true,
    "name": "Under The Bridge",
    "popularity": 36,
    "preview_url": "https://p.scdn.co/mp3-preview/5c052c976f9ad6610a1e8daf79e467e1e3943632?cid=2e581de7a4de43ea90e2ede13f73f090",
    "track_number": 11,
    "type": "track",
    "uri": "spotify:track:1MMlmA1APa0MevW3Bm2PpE",
  },
  Object {
    "album": Object {
      "album_type": "album",
      "artists": Array [
        Object {
          "external_urls": Object {
            "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
          },
          "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
          "id": "0L8ExT028jH3ddEcZwqJJ5",
          "name": "Red Hot Chili Peppers",
          "type": "artist",
          "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
        },
      ],
      "external_urls": Object {
        "spotify": "https://open.spotify.com/album/1p43Uwq9mU8WCF6wQR0XFq",
      },
      "href": "https://api.spotify.com/v1/albums/1p43Uwq9mU8WCF6wQR0XFq",
      "id": "1p43Uwq9mU8WCF6wQR0XFq",
      "images": Array [
        Object {
          "height": 640,
          "url": "https://i.scdn.co/image/27e5e75471c86e8a956e3113e4e09e304d9a22f7",
          "width": 640,
        },
        Object {
          "height": 300,
          "url": "https://i.scdn.co/image/b2ea7a89024b2824640bd6b2d132d442c4d071d0",
          "width": 300,
        },
        Object {
          "height": 64,
          "url": "https://i.scdn.co/image/dad1d7e30b3f1e29d5ff47d1f9cdb1bd11056485",
          "width": 64,
        },
      ],
      "name": "Woodstock 1994 (Live)",
      "release_date": "2015-01-26",
      "release_date_precision": "day",
      "type": "album",
      "uri": "spotify:album:1p43Uwq9mU8WCF6wQR0XFq",
    },
    "artists": Array [
      Object {
        "external_urls": Object {
          "spotify": "https://open.spotify.com/artist/0L8ExT028jH3ddEcZwqJJ5",
        },
        "href": "https://api.spotify.com/v1/artists/0L8ExT028jH3ddEcZwqJJ5",
        "id": "0L8ExT028jH3ddEcZwqJJ5",
        "name": "Red Hot Chili Peppers",
        "type": "artist",
        "uri": "spotify:artist:0L8ExT028jH3ddEcZwqJJ5",
      },
    ],
    "disc_number": 1,
    "duration_ms": 290520,
    "explicit": false,
    "external_ids": Object {
      "isrc": "QM6MZ1498870",
    },
    "external_urls": Object {
      "spotify": "https://open.spotify.com/track/44ypSpvvjNlGBMJcCh3ke5",
    },
    "href": "https://api.spotify.com/v1/tracks/44ypSpvvjNlGBMJcCh3ke5",
    "id": "44ypSpvvjNlGBMJcCh3ke5",
    "is_local": false,
    "is_playable": true,
    "name": "Under the Bridge (Live)",
    "popularity": 17,
    "preview_url": "https://p.scdn.co/mp3-preview/22edcc99b1afa25a39c01f1361b70760284695dc?cid=2e581de7a4de43ea90e2ede13f73f090",
    "track_number": 13,
    "type": "track",
    "uri": "spotify:track:44ypSpvvjNlGBMJcCh3ke5",
  },
]
`;
